rfcs/syntax-floor-division-operator.md

```diff
@@ -0,0 +1,60 @@
+# Floor division operator
+
+## Summary
+
+Add floor division operator `//` to ease computing with integers.
+
+## Motivation
+
+Integers are everywhere. Indices, pixel coordinates, offsets, ranges, quantities, counters, rationals, fixed point arithmetic and bitwise operations all use integers.
+
+Luau is generally well suited to work with integers. The math operators +, -, \*, ^ and % support integers. That is, given integer operands these operators produce an integer result (provided that the result fits into representable range of integers). However, that is not the case with the division operator `/` which in the general case produces numbers with fractionals.
+
+To overcome this, typical Luau code performing integer computations needs to wrap the result of division inside a call to `math.floor`. This has a number of issues and can be error prone in practice.
+
+A typical mistake is to forget to use `math.floor`. This can produce subtle issues ranging from slightly wrong results to program crashes. A crash could occur, for example, when the result of division is used to fetch from a table with only integer keys, which produces nil and the program crashes soon after. Another type of crash occurs when an accidental fractional number is passed to a C function that expects integers. Particularly problematic is incorrect code which seems to work with frequently used data, only to fail with some rare input. For example, image sizes often have power of two dimensions, so code dealing with them may appear to work fine until much later some rare image has an odd size and a division by two in the code does not produce the correct result. Due to better ergonomics of the floor division operator, it becomes a second nature to write `//` everywhere when integers are involved and thus this class of bugs is much less likely to happen.
```

Comment:
nit: this should ideally use "script errors" instead of "program crashes" for cases like `nil` result of a table lookup; similarly, a C function receiving a fractional number would result in a rounded result as long as `lua_tointeger` is used, which can result in a logic error but hopefully doesn't actually crash the host!

(this is not to say that the concerns with accidentally forgetting `math.floor` are unfounded, I'd just rather use more precise language in RFCs)

Reply:
Fair enough. (I usually use "crash" as a synonym for script errors because in my domain -- games -- a script error brings down the application from users' POV). I've modified the paragraph based on your feedback.